packages/@react-stately/combobox/src/useComboBoxState.ts

```diff
@@ -79,27 +84,56 @@ export function useComboBoxState<T extends object>(props: ComboBoxStateProps<T>)
     items: props.items ?? props.defaultItems
   });
 
+  // This is only used if items are not controlled
+  let originalCollection = collection;
+
   let filteredCollection = useMemo(() => (
     // No default filter if items are controlled.
     props.items != null || !defaultFilter
       ? collection
       : filterCollection(collection, inputValue, defaultFilter)
   ), [collection, inputValue, defaultFilter, props.items]);
 
-  let triggerState = useMenuTriggerState(props);
-  let open = (focusStrategy?: FocusStrategy) => {
+  // Track what action is attempting to open the menu
+  let menuOpenTrigger = useRef('focus' as MenuTriggerAction);
+  let onOpenChange = (open: boolean) => {
+    if (props.onOpenChange) {
+      props.onOpenChange(open, open ? menuOpenTrigger.current : undefined);
```

Comment:
Seems a little hacky to use a ref for this... I guess the other option would be to not use menu trigger state at all and reimplement it and that seems possibly worse 🤔

Reply:
yeah def a bit hacky, but as you said the alternative would be to reimplement the open state logic from `useMenuTriggerState`+`useOverlayTriggerState` just so we can get onOpenChange to return the menu trigger action as well. I figured changing those hooks themselves to track menu trigger action was undesirable as well.